flood_adapt/object_model/direct_impacts.py

```diff
@@ -373,14 +374,24 @@ def _create_infometrics(self, fiat_results_path):
         metrics_writer.parse_metrics_to_file(
             df_results=df, metrics_path=metrics_outputs_path, write_aggregate=None
         )
-        metrics_writer.parse_metrics_to_file(
-            df_results=df, metrics_path=metrics_outputs_path, write_aggregate="all"
-        )
 
-    def _create_infographics(self):
+        if self.hazard.event_mode != "risk":
+            metrics_writer.parse_metrics_to_file(
+                df_results=df, metrics_path=metrics_outputs_path, write_aggregate="all"
+            )
+
+    def _create_infographics(self, mode):
         # Get the infographic
-        InfographicsParser().write_infographics_to_file(
-            scenario_name=self.name,
-            database_path=Path(self.database_input_path).parent,
-            keep_metrics_file=True,
+        database_path = Path(self.database_input_path).parent
+        metrics_path = database_path.joinpath(
+            "output", "infometrics", f"{self.name}_metrics.csv"
         )
```

Comment:
couldn't this be saved in the class when _create_infometrics is run? This way we make sure that the path is defined once.